routes/volunteers_routes.js

```diff
@@ -0,0 +1,251 @@
+    return Volunteer.get_by_user(user_id).then((vol_data) => {
+        if (vol_data.get('role_id') <= perm_level) {
+            next();
+        } else {
+            err('No Permissions');
+        }
+    });
+};
+//GET /volunteer/volunteers
+var get_volunteers = function(req, res) {
+    __has_permissions(req.user.id, 0, () => {
+        //var user_ids = [1]; //find user by name or email and by ticket.
+        /*
+        Volunteer.query((qb) => {
+            if (req.query.deps !== undefined) {
+                qb.whereIn('department_id', req.query.deps);
+            }
+            if (req.query.roles !== undefined) {
+                qb.whereIn('role_id', req.query.roles);
+            }
+        }).fetchAll().then((vols) => {
+            if (vols === null) {
+                res.json('[]')
+            }
+            
+            res.json(vols.toJSON());
+        }).catch((err) => {
+            res.status(500).json({
+                error: true,
+                data: {
+                    message: err.message
+                }
+            });
+        });
+        */
+        DrupalAccess.get_user_by_email(req.query.email).then((user) => {
+            Volunteer.query((qb) => {
+                qb.where('user_id', user.id);
+                if (req.query.deps !== undefined) {
+                    qb.whereIn('department_id', req.query.deps);
+                }
+                if (req.query.roles !== undefined) {
+                    qb.whereIn('role_id', req.query.roles);
+                }
+            }).fetchAll().then((vols) => {
+                if (vols === null) {
+                    res.json('[]')
+                }
+                var ret = _.map(vols.models, (x) => {
+                    return {
+                        user_id: user.id,
+                        email: user.email,
+                        permission: x.role_id,
+                        department_id: x.department_id,
+                        first_name: user.first_name,
+                        last_name: user.last_name,
+                        phone_number: user.phone_number,
+                        got_ticket: true,
+                        comment: "from Volunteers table"
+                    };
+                });
+
+                res.json(ret);
+            }).catch((err) => {
+                res.status(500).json({
+                    error: true,
+                    data: {
+                        message: err.message
+                    }
+                });
+            });
+        });
+    }, (err) => {
+        res.status(401).json({ message: err });
+    });
+};
+//GET /volunteer/department/:department_id/volunteers
+var get_department_volunteers = function(req, res) {
+    __has_permissions(req.user.id, 2, () => {
+
+        DrupalAccess.get_user_by_email(req.query.email).then((user) => {
+            Volunteer.query((qb) => {
+                qb.where('user_id', user.id);
```

Comment:
if i'm reading this correctly this fetches the departments for a given, user - shouldn't this return all departments independent of user?